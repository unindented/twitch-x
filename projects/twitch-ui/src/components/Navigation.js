import React from 'react'
import styled from 'styled-components'
import NavigationLogo from './NavigationLogo'
import NavigationLinks from './NavigationLinks'

const NavigationRoot = styled.nav`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  display: flex;
  overflow: hidden;
  width: ${props => props.theme.layout.navCollapsedWidth};
  flex-direction: column;
  background-color: ${props => props.theme.colors.quaternary};
  color: ${props => props.theme.colors.white};
  transition: width 0.2s ease-in-out;

  &:hover {
    width: ${props => props.theme.layout.navExpandedWidth};
  }

  *[dir='rtl'] & {
    right: 0;
    left: auto;
  }
`

export default function Navigation () {
  return (
    <NavigationRoot>
      <NavigationLogo />
      <NavigationLinks />
    </NavigationRoot>
  )
}
