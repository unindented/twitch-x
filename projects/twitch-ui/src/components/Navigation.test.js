import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import Navigation from './Navigation'

describe('Navigation', () => {
  let tree

  beforeEach(() => {
    const renderer = new ShallowRenderer()
    renderer.render(<Navigation />)
    tree = renderer.getRenderOutput()
  })

  it('renders', () => {
    expect(tree).toMatchSnapshot()
  })
})
