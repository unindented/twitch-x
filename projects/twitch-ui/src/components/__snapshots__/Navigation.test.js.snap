// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders 1`] = `
<Navigation__NavigationRoot>
  <NavigationLogo />
  <NavigationLinks />
</Navigation__NavigationRoot>
`;
